featurebyte/worker/task/catalog_cleanup.py

```diff
@@ -0,0 +1,230 @@
+        self.feature_store_service = feature_store_service
+        self.all_catalog_service = all_catalog_service
+        self.persistent = persistent
+        self.storage = storage
+        self.session_manager_service = session_manager_service
+
+    @property
+    def model_packages(self) -> list[Any]:
+        """
+        Model packages to look for catalog specific models
+
+        Returns
+        -------
+        list[Any]
+            list of model packages
+        """
+        return [featurebyte.models]
+
+    @property
+    def remap_model_classes(self) -> dict[str, Type[FeatureByteCatalogBaseDocumentModel]]:
+        """
+        Remap model classes to cleanup
+
+        Returns
+        -------
+        dict[str, Type[FeatureByteCatalogBaseDocumentModel]]
+            remap model classes
+        """
+        return {"table": ProxyTableModel}
+
+    @property
+    def catalog_specific_model_class_pairs(
+        self,
+    ) -> list[tuple[str, Type[FeatureByteCatalogBaseDocumentModel]]]:
+        """
+        Catalog specific model class pairs to cleanup
+
+        Returns
+        -------
+        list[Type[FeatureByteCatalogBaseDocumentModel]]
+            list of catalog specific models
+        """
+        collection_name_to_model: dict[str, Any] = {}
+        for package in self.model_packages:
+            for loader, module_name, is_pkg in pkgutil.walk_packages(
+                package.__path__, package.__name__ + "."
+            ):
+                module = importlib.import_module(module_name)
+                for _class in get_catalog_subclasses_in_module(module):
+                    collection_name_to_model[_class.collection_name()] = _class
+
+        # remap the model classes to override the class used for the collection
+        collection_name_to_model.update(self.remap_model_classes)
+        return list(collection_name_to_model.items())
+
+    async def get_task_description(self, payload: CatalogCleanupTaskPayload) -> str:
+        return "Catalog clean up task"
+
+    async def _cleanup_warehouse_tables(
+        self, catalog: CatalogModel, warehouse_tables: set[TableDetails]
+    ) -> None:
+        feature_store = await self.feature_store_service.get_document(
+            catalog.default_feature_store_ids[0]
+        )
+        session = await self.session_manager_service.get_feature_store_session(
+            feature_store, user_override=User(id=feature_store.user_id)
+        )
+        fs_source_info = feature_store.get_source_info()
+        for warehouse_table in warehouse_tables:
+            try:
+                await session.drop_table(
+                    table_name=warehouse_table.table_name,
+                    schema_name=warehouse_table.schema_name or fs_source_info.schema_name,
+                    database_name=warehouse_table.database_name or fs_source_info.database_name,
+                )
+            except DataWarehouseOperationError:
+                pass
```

Comment:
Would it be useful to add logging here since in most cases errors are not expected?